engine/src/main/java/org/terasology/math/VectorPools.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.math;
+
+import org.terasology.math.geom.Vector3i;
+
+public class VectorPools {
```

Comment:
The class should be named `VectorPool`, without the s.

(it's SolShip, never SolShips - unless it is a collection of SolShips. In this case, this is a collection of vectors, not a collection of vectorPools, so..)

Reply:
Also, let's make this class more generic , that can be mixed with Templates (so that we can have `VectorPool<Vector3i>` or `VectorPool<3f>`) or inherited (to get `VectorPool3i extends VectorPool<Vector3i>` or `VectorPool3f extends VectorPool<Vector3f>`)? Makes the class way more extensible and flexible.